cmake/Tools/layout_tool.py

```diff
@@ -244,7 +244,11 @@ def copy_asset_files_to_layout(project_asset_folder, target_platform, layout_tar
                             abs_dst)
             continue
 
-        if os.path.isfile(abs_dst):
+        # If the target file doesn't exist, copy it
+        if not os.path.exists(abs_dst):
+            logging.debug("Copying %s -> %s", abs_src, abs_dst)
+            shutil.copy2(abs_src, abs_dst)
+        elif os.path.isfile(abs_dst):
```

Comment:
This is the fix so `copy_asset_files_to_layout` does actually copy the files in the root folder.

@amzn-sj are you familiarized with this file? Is there any side-effect problems with this?

Reply:
I think it's ok. @spham-amzn would also know about this.

Reply:
For VFS, the bootstrap setreg and engine.json need to be copied to the sdcard. If not, it's going to assume the assets are in the APK.

Reply:
Maybe the solution is to copy only the .setreg files in the root cache folder, not all the files in the root... We have to figure out what's the bare minimum data that the initialization needs to know it's a VFS connection, the ip, the port (all this in the .setreg files from the bootstrap). It needs to know a project folder to generate a AP token, otherwise if the token send for the connection doesn't match what's expected by AP a negotiation failure happens.

So this means that Android needs to configure itself to read .setreg files from device and then when it realizes it needs VFS then switch whatever it needs (e.g. aliases) to be generate valid asset requests for VFS.

Reply:
Sure. But those files would still need to go into the <sdcard_path>/com.o3de.project/files directory instead of going into the APK I think.

Reply:
I see, I'll investigate how to copy these files into the sdcard instead of the apk. Thanks.

Reply:
Just for testing I copied the root cache files to `/sdcard/Android/data/com.o3de.project/files` myself manually. It was able to connect to AP correctly, so that's a good start, but it decides to use `/sdcard/Android/data/com.o3de.project/files` as the main folder instead of VFS for some reason as it prints:

````
Application is configured to use device local files at /sdcard/Android/data/com.o3de.project/files
````

If that's printed it's because `remoteFileSystemEnabled` was false here for some reason:

````
bool remoteFileSystemEnabled{};
        AZ::SettingsRegistryMergeUtils::PlatformGet(*settingsRegistry, remoteFileSystemEnabled,
            AZ::SettingsRegistryMergeUtils::BootstrapSettingsRootKey, "remote_filesystem");
        if (remoteFileSystemEnabled)
````

I'll keep looking, that should have returned true.